2-Browsers/Week1/homework/ex5-catWalk/index.js

```diff
@@ -15,8 +16,50 @@
    for 5 seconds, and then replace the img with the original image and have it 
    continue the walk.
 -----------------------------------------------------------------------------*/
+//crating img and changing style of img
+const img = document.getElementsByTagName('img')[0]; // [0] i used it for first desired image, because it returns an array and i want to take first image
+img.style.left = '0px';
+
+const dancingCat = 'https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif?itemid=10561424'
+const originalImgSrc = img.src;
+const originalImgWidth = img.width;
+let walkForwards = true;
+
+//moving cat to right 10px
 function catWalk() {
-  // TODO complete this function
+
+  let currentLeft = parseInt(img.style.left); //converting px to int number
+  const middlePosition = (window.innerWidth - originalImgWidth) / 2;
+  // The innerWidth property returns the width of a window's content area
+  if (walkForwards && (currentLeft > (window.innerWidth-img.width))) {
+    walkForwards = false;
+  }
+
+  if (!walkForwards && (currentLeft <= 0)) {
+    walkForwards = true;
+  }
+  
+  if (walkForwards) {
+    img.style.left = (currentLeft + 10) + 'px';
+  } 
+
+  else {
+    img.style.left = (currentLeft - 10) + 'px';
+  }
+
+  if (currentLeft >= middlePosition && currentLeft <= middlePosition ) {
```

Comment:
This will only ever trigger when the `currentLeft` is exactly equal to `middlePosition`. Keep in mind that the original assignment doesn't require you to make the cat moonwalk backwards (though it's very cool!), and the solution here is simpler if you don't need to do that.

If you do want to do that, you'll want to make sure you check for something that you're sure it'll actually equal. Now, the exact half of the screen can end in single digits, but the cat will only ever get to a multiple of ten. So you can just round off the middle position to the nearest multiple of ten: `currentLeft === Math.round(middlePosition / 10) * 10`

You'll still have to solve it getting stuck on its way back, but that should be doable by making sure the inside of the `setTimeout` is correct :)

Reply:
I changed by deleting backward part and fixed the gif's current position.